In [4]:
import pandas as pd
import numpy as np
import os

In [5]:
ls

 C 드라이브의 볼륨: OS
 볼륨 일련 번호: 1411-45FD

 C:\workspace\WASSUP2\WASSUP2\03_Data_Preprocessing 디렉터리

2024-03-25  오후 11:58    <DIR>          .
2024-03-19  오전 11:32    <DIR>          ..
2024-03-25  오후 10:07    <DIR>          .ipynb_checkpoints
2024-03-24  오전 10:34           512,795 01_Numpy.ipynb
2024-03-23  오후 01:52            20,092 01_Numpy손코딩.ipynb
2024-03-24  오전 10:17            29,778 02_Numpy_Quiz_실습용.ipynb
2024-03-21  오전 09:11            21,861 02_Numpy_Quiz_정답.ipynb
2024-03-24  오후 06:22           118,529 03.01_Pandas 패키지의 소개.ipynb
2024-03-24  오후 07:06            54,853 03.02_Pandas 데이터 입출력.ipynb
2024-03-25  오후 11:38            56,167 03.03_Pandas 데이터프레임 고급 인덱싱.ipynb
2024-03-24  오후 07:59           231,363 03.04_Pandas 데이터프레임 데이터 조작.ipynb
2024-03-25  오후 11:11           213,218 03.05_Pandas 데이터프레임 인덱스 조작.ipynb
2024-03-25  오후 10:11           188,751 03.05_Pandas 데이터프레임 인덱스 조작확인차.ipynb
2024-03-25  오후 11:45           125,454 03.06_Pandas 데이터프레임 합성.ipynb
2024-03-25  오전 11:52           257,9

# Q1-1

```
아래의 파일을 이용하여 df_covid 데이타프레임을 생성하여라.
   
seoul-covid19_6_30_.csv    

```

In [13]:
df_covid = pd.read_csv('./data/seoul-covid19_6_30_.csv')
df_covid 

연번     환자    확진일   거주지     여행력          접촉력         조치사항
0     1321  12830  6.30.  영등포구       -  영등포구 확진자 접촉            -
1     1320  12829  6.30.   관악구       -         확인 중        보라매병원
2     1319  12831  6.30.  영등포구       -         확인 중      중앙대학교병원
3     1318  12808  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
4     1317  12809  6.30.   관악구       -   강남구 역삼동 모임        보라매병원
...    ...    ...    ...   ...     ...          ...          ...
1316     5      9  1.31.   성북구       -        #5 접촉    서울의료원(퇴원)
1317     4      7  1.30.   마포구  중국 우한시        해외 접촉    서울의료원(퇴원)
1318     3      6  1.30.   종로구       -        #3 접촉  서울대학교병원(퇴원)
1319     2      5  1.30.   중랑구  중국 우한시        해외 접촉    서울의료원(퇴원)
1320     1      2  1.24.   강서구  중국 우한시        해외 접촉  국립중앙의료원(퇴원)

[1321 rows x 7 columns]

# Q1-2
```
'연번' 컬럼은 삭제하고  '환자' 컬럼 값 기준으로 정렬한 후 '환자' 컬럼을 인덱스로 설정하여라. 
```

In [14]:
df_covid = df_covid.drop("연번", axis=1)

In [15]:
df_covid = df_covid.sort_values(by="환자").set_index("환자")
df_covid.head()

확진일  거주지     여행력    접촉력         조치사항
환자                                        
2   1.24.  강서구  중국 우한시  해외 접촉  국립중앙의료원(퇴원)
5   1.30.  중랑구  중국 우한시  해외 접촉    서울의료원(퇴원)
6   1.30.  종로구       -  #3 접촉  서울대학교병원(퇴원)
7   1.30.  마포구  중국 우한시  해외 접촉    서울의료원(퇴원)
9   1.31.  성북구       -  #5 접촉    서울의료원(퇴원)

# Q1-3
어느 지역에서 제일 많은 확진자가 발생했는지 1~5위 까지 출력하여라 (거주지 기준)

In [17]:
df_group = df_covid.groupby("거주지").count().sort_values(by="확진일", ascending=False)
df_group

확진일  여행력  접촉력  조치사항
거주지                      
관악구   117  117  117   117
구로구    86   86   86    86
강남구    83   83   83    83
강서구    78   78   78    78
양천구    70   70   70    70
영등포구   65   65   65    65
도봉구    62   62   62    62
서초구    56   56   56    56
송파구    55   55   55    55
은평구    53   53   53    53
동작구    53   53   53    53
용산구    50   50   50    50
성동구    49   49   49    49
노원구    43   43   43    43
마포구    39   39   39    39
동대문구   38   38   38    38
강동구    35   35   35    35
서대문구   35   35   35    35
성북구    33   33   33    33
중랑구    33   33   33    33
금천구    32   32   32    32
강북구    24   24   24    24
종로구    22   22   22    22
광진구    17   17   17    17
중구     12   12   12    12
광명시     8    8    8     8
고양시     7    7    7     7
남양주     5    5    5     5
성남시     5    5    5     5
김포시     5    5    5     5
인천시     5    5    5     5
스페인     4    4    4     4
양평      3    3    3     3
부천시     3    3    3     3
양주시     3    3    3     3
수원시     3    3    3     3
의정부     2    2    2     2
용인시     2    2    2     2
구리시     2    2    2     2
대구시     2    2    2     2
천안시     2    2    2     2
대전시     2    2    2     2
재외국민    1    1    1     1
주소불명    1    1    1     1
파주시     1    1    1     1
의왕시     1    1    1     1
평택시     1    1    1     1
아산시     1    1    1     1
경북      1    1    1     1
양산시     1    1    1     1
안양시     1    1    1     1
안양      1    1    1     1
안산시     1    1    1     1
기타      1    1    1     1
시흥시     1    1    1     1
부평      1    1    1     1
미국      1    1    1     1
국외거주    1    1    1     1
군포시     1    1    1     1
화성시     1    1    1     1

In [19]:
top5=list(df_group["확진일"].nlargest(5, keep="all").index.values)
top5

['관악구', '구로구', '강남구', '강서구', '양천구']


# Q1-4
확진자가 가장 많이 발생한 1, 2, 3위 지역의 데이타만 출력하여라 

In [20]:
for i in top5[:3]:
    print(df_covid[df_covid["거주지"]==i].sort_values(by="거주지"))

         확진일  거주지  여행력         접촉력        조치사항
환자                                            
935    2.25.  관악구    -         확인중  삼육서울병원(퇴원)
12516  6.23.  관악구    -  동작구 확진자 접촉       보라매병원
12504  6.23.  관악구    -        확인 중        서남병원
12395  6.20.  관악구    -        확인 중       서울의료원
12271  6.18.  관악구    -     리치웨이 관련   보라매병원(퇴원)
...      ...  ...  ...         ...         ...
9879   3.31.  관악구    -   구로구 교회 관련    서남병원(퇴원)
9876   3.31.  관악구    -   구로구 교회 관련   서울의료원(퇴원)
9837   3.31.  관악구    -   구로구 교회 관련    서남병원(퇴원)
10749  4.27.  관악구  필리핀    해외 접촉 추정   적십자병원(퇴원)
12829  6.30.  관악구    -        확인 중       보라매병원

[117 rows x 5 columns]
         확진일  거주지  여행력          접촉력         조치사항
환자                                              
348    2.22.  구로구    -   타시도 확진자 접촉  국립중앙의료원(퇴원)
11842  6.08.  구로구    -      리치웨이 관련    보라매병원(퇴원)
11841  6.08.  구로구    -      리치웨이 관련      국립중앙의료원
11840  6.08.  구로구    -  수도권 개척교회 관련         서남병원
11839  6.08.  구로구    -      리치웨이 관련    보라매병원(퇴원)
...      ...  ...  ...

# Q1-5
'확진일' 컬럼값을 아래와 같은 형태로 변경하여라
<br>
'1.24.' => '2020-01-24'

# Q1-6
'확진일' 컬럼을 시계열 데이타로 형태를 변경하여라 

# Q1-7
3월에 발생한 확진자의 데이타만 출력하여라.

# Q1-8
주말에 발생한 확진자의 데이타만 출력하여라.

# Q1-9
'콜센터'와 관련된 확진자의 데이타를 출력하여라. 

## Q1-10
'우한'과 관련된 확진자의 데이타만 출력하여라. 

## Q1-11
사망자는 모두 몇명인가?

# Q2-1

아래의 파일을 이용하여 데이타프레임을 생성하여라.<br>
   
전국 평균 분양가격.csv 

# Q2-2

데이터프레임의 행과 열을 서로 변경하여라.

# Q2-3

1행을 삭제하고 2행을 컬럼으로 지정하여라. 

# Q2-4

'지역' 컬럼을 시계열 데이타로 변경하고 인덱스로 설정하여라. 

# Q2-5

2015년 6월 이후의 데이타만 출력하여라 

# Q2-6

2014년도의 서울, 인천 경기 컬럼만 출력하여라. 

# Q2-7
csv 파일 형태로 현재의 데이타프레임을 저장하여라. 
<br>
output/평균분양가격_서울인천경기.csv